### ISSUE TITLE
Some examples in tests/integrate failed on Sugon platform

### ISSUE THREAD

**Issue by LiuXiaohui123321** [deepmodeling/abacus-develop#1013]
When running examples in tests/integrate using ABACUS [v2.2.3](https://github.com/abacusmodeling/abacus-develop/releases/tag/v2.2.3) on Sugon platform, there are some examples failed.
For more information, please see the attached file.
[abacus v2.2.3自动测试集测试结果.md](https://github.com/deepmodeling/abacus-develop/files/8888982/abacus.v2.2.3.md)

**Comment by LiuXiaohui123321** [comment]
[abacus v2.2.3自动测试集测试结果.pdf](https://github.com/deepmodeling/abacus-develop/files/8889011/abacus.v2.2.3.pdf)

**Comment by dyzheng** [comment]
Thanks for your testing, we will try to reproduce these bugs which you reported.

**Comment by caic99** [comment]
Hi @LiuXiaohui123321 ,
It's difficult to locate errors on sugon platform since we don't have access to it. Would you try the following steps for us to locate the problem?

- Use module `compiler/intel/oneapi-2021.3` to compile abacus
- Set env var `OMP_NUM_THREADS=1` when running abacus
- Compile and run the unit tests of abacus

**Comment by LiuXiaohui123321** [comment]
OK! If change to use the intel-oneapi , whether the elpa needs to recompile? And what does "the unit tests of abacus" mean? Is it "tests/integrate "? @caic99

**Comment by caic99** [comment]
> OK! If change to use the intel-oneapi , whether the elpa needs to recompile? 

I think there won't be much problem - only recompiling abacus would be enough.

> And what does "the unit tests of abacus" mean? Is it "tests/integrate "? @caic99

No. The unit tests are some independent tests of the functions in abacus. Please install googletest
```bash
 git clone https://github.com/google/googletest.git \
    && cd googletest && cmake . && make install
```

and build abacus with 
```bash
cmake -B build -DBUILD_TESTING=ON
cmake --build build -j8
cmake --install build
cmake --build build --target test ARGS="-V"
```

**Comment by LiuXiaohui123321** [comment]
Update: ABACUS [v2.2.4](https://github.com/abacusmodeling/abacus-develop/releases/tag/v2.2.4) has the some problems with ABACUS [v2.2.3](https://github.com/abacusmodeling/abacus-develop/releases/tag/v2.2.3). @dyzheng @caic99

**Comment by LiuXiaohui123321** [comment]
> > OK! If change to use the intel-oneapi , whether the elpa needs to recompile?
> 
> I think there won't be much problem - only recompiling abacus would be enough.
> 
> > And what does "the unit tests of abacus" mean? Is it "tests/integrate "? @caic99
> 
> No. The unit tests are some independent tests of the functions in abacus. Please install googletest
> 
> ```shell
>  git clone https://github.com/google/googletest.git \
>     && cd googletest && cmake . && make install
> ```
> 
> and build abacus with
> 
> ```shell
> cmake -B build -DBUILD_TESTING=ON
> cmake --build build -j8
> cmake --install build
> cmake --build build --target test ARGS="-V"
> ```

Thanks! And I'll try it.

**Comment by LiuXiaohui123321** [comment]
> ```shell
>  git clone https://github.com/google/googletest.git \
>     && cd googletest && cmake . && make install
> ```
In the last step to install googletest, there was an error:
> Install the project...
-- Install configuration: ""
-- Up-to-date: /usr/local/include
CMake Error at googlemock/cmake_install.cmake:46 (file):
  file INSTALL cannot set permissions on "/usr/local/include": Operation not
  permitted.
Call Stack (most recent call first):
  cmake_install.cmake:47 (include)

It seems to should specify the install directory. @caic99

**Comment by caic99** [comment]
@LiuXiaohui123321 It is possible to specify a different installation directory by adding -DCMAKE_INSTALL_PREFIX=/path/to/install/dir to the CMake command line.

**Comment by LiuXiaohui123321** [comment]
@caic99 
I've installed googletest,. But when I try to install abacus as you suggested, it gives some errors about googletest.
It seems like that the googletest directory needs to be in somewhere  right, or it will download it from github.

> -- Found IntelMKL: /public/software/compiler/intel/onepai-2021.3/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so  
-- Could NOT find GTest (missing: GTest_DIR)
[ 11%] Creating directories for 'googletest-populate'
[ 22%] Performing download step (git clone) for 'googletest-populate'
Cloning into 'googletest-src'...
fatal: unable to access 'https://github.com/google/googletest.git/': Failed connect to github.com:443; Connection timed out

**Comment by caic99** [comment]
> GTest_DIR

Hi @LiuXiaohui123321 ,
From the prompted message, would you try define -DGTest_DIR=xxx?

**Comment by LiuXiaohui123321** [comment]
> Hi @LiuXiaohui123321 , It's difficult to locate errors on sugon platform since we don't have access to it. Would you try the following steps for us to locate the problem?
> 
> * Use module `compiler/intel/oneapi-2021.3` to compile abacus
> * Set env var `OMP_NUM_THREADS=1` when running abacus
> * Compile and run the unit tests of abacus

@dyzheng @caic99 The unit tests have finished, and the result is as follows:
> 88% tests passed, 7 tests failed out of 60
Total Test time (real) = 4175.05 sec
The following tests FAILED:
         38 - md_LJ_pot (Failed)
         40 - md_fire (Failed)
         41 - md_nve (Failed)
         42 - md_nvt_ads (Failed)
         43 - md_nvt_nhc (Failed)
         44 - md_msst (Failed)
         45 - md_lgv (Failed)

**Comment by LiuXiaohui123321** [comment]
Update: I try to use intel oneapi-2021.3 to run examples in tests/integrate (ABACUS [v2.2.4](https://github.com/abacusmodeling/abacus-develop/releases/tag/v2.2.4)), and most examples finish normally.
>  [  FAILED  ]  1 test cases out of 623 failed.
211_NO_S2_elec_add

So what is the difference between using intel oneapi-2021.3 and using intel 2017.4.239 or other compiler version?
Is there any special treatment for intel oneapi-2021.3 when writing or refactoring programs?
@caic99 @dyzheng

**Comment by caic99** [comment]
> @dyzheng @caic99 The unit tests have finished, and the result is as follows:
> 
> > 88% tests passed, 7 tests failed out of 60
> > Total Test time (real) = 4175.05 sec
> > The following tests FAILED:
> > 38 - md_LJ_pot (Failed)
> > 40 - md_fire (Failed)
> > 41 - md_nve (Failed)
> > 42 - md_nvt_ads (Failed)
> > 43 - md_nvt_nhc (Failed)
> > 44 - md_msst (Failed)
> > 45 - md_lgv (Failed)

Hi @LiuXiaohui123321,
This seems to be a md-related issue. Would you provide the output detail of these unit tests for us? You can run md tests by: `cd build && ctest -R md -V`.


> Update: I try to use intel oneapi-2021.3 to run examples in tests/integrate (ABACUS [v2.2.4](https://github.com/abacusmodeling/abacus-develop/releases/tag/v2.2.4)), and most examples finish normally.
> 
> > [  FAILED  ]  1 test cases out of 623 failed.
> > 211_NO_S2_elec_add
> 

Please refer to https://github.com/deepmodeling/abacus-develop/issues/945. Personally, I think this issue is related to OpenMP, but not the version of Intel compiler.

> So what is the difference between using intel oneapi-2021.3 and using intel 2017.4.239 or other compiler version? Is there any special treatment for intel oneapi-2021.3 when writing or refactoring programs? @caic99 @dyzheng

There are no special treatment between different Intel compilers. 
As far as I know, when we are migrating elpa to Intel oneAPI compilers, we met some problems at first, but things should go smoothly now.
The difference between Intel compilers and other compilers lies in linking phase of ABACUS. We assuming the user is compiling elpa with the same set of compilers compiling ABACUS, and add the corresponding fortran library flags for it - `gfortran` of `ifcore`. However, this is not perfect, because the user might choose a different compiler for elpa and ABACUS. We'll refine this process later.

**Comment by mohanchen** [comment]
Since ABACUS has released 3.2 version, we will discuss this if new version met problems.
